neo/Core/Blockchain.cs

```diff
@@ -341,24 +344,95 @@ public ECPoint[] GetValidators()
 
         public virtual IEnumerable<ECPoint> GetValidators(IEnumerable<Transaction> others)
         {
-            //TODO: 此处排序可能将耗费大量内存，考虑是否采用其它机制
-            VoteState[] votes = GetVotes(others).OrderBy(p => p.PublicKeys.Length).ToArray();
-            int validators_count = (int)votes.WeightedFilter(0.25, 0.75, p => p.Count.GetData(), (p, w) => new
-            {
-                ValidatorsCount = p.PublicKeys.Length,
-                Weight = w
-            }).WeightedAverage(p => p.ValidatorsCount, p => p.Weight);
-            validators_count = Math.Max(validators_count, StandbyValidators.Length);
-            Dictionary<ECPoint, Fixed8> validators = GetEnrollments().ToDictionary(p => p.PublicKey, p => Fixed8.Zero);
-            foreach (var vote in votes)
+            DataCache<UInt160, AccountState> accounts = GetStates<UInt160, AccountState>();
+            DataCache<ECPoint, ValidatorState> validators = GetStates<ECPoint, ValidatorState>();
+            MetaDataCache<ValidatorsCountState> validators_count = GetMetaData<ValidatorsCountState>();
+            foreach (Transaction tx in others)
             {
-                foreach (ECPoint pubkey in vote.PublicKeys.Take(validators_count))
+                foreach (TransactionOutput output in tx.Outputs)
+                {
+                    AccountState account = accounts.GetAndChange(output.ScriptHash, () => new AccountState(output.ScriptHash));
+                    if (account.Balances.ContainsKey(output.AssetId))
+                        account.Balances[output.AssetId] += output.Value;
+                    else
+                        account.Balances[output.AssetId] = output.Value;
+                    if (output.AssetId.Equals(GoverningToken.Hash) && account.Votes.Length > 0)
+                    {
+                        foreach (ECPoint pubkey in account.Votes)
+                            validators.GetAndChange(pubkey, () => new ValidatorState(pubkey)).Votes += output.Value;
+                        validators_count.GetAndChange().Votes[account.Votes.Length - 1] += output.Value;
```

Comment:
`account.Votes` ever have the same length than `GetAndChange().Votes` ?

Reply:
The length is always equals to `Blockchain.MaxValidators`.